torchdynamo/_guards.cpp

```diff
@@ -52,7 +61,30 @@ class TensorCheck {
       const auto &sizes = v.sizes();
       const auto &strides = v.strides();
       for (size_t i = 0; i < ndim; ++i) {
-        if (sizes_[i] != sizes[i] || strides_[i] != strides[i]) {
+        if (sizes_[i] != sizes[i]) {
+          bool is_dynamic_dim = false;
+          for (int dd = 0; dd < dynamic_dims_.size(); dd++) {
```

Comment:
This is `O(n^2)`.  Since guard checking is on the hot path, we should just a `std::vector<bool> _is_dynamic(ndim, False)` so we can do checks for dynamic without a loop.

Reply:
Yeah, this is just a hack to get us going, will change.